ksqldb-engine/src/main/java/io/confluent/ksql/engine/EngineContext.java

```diff
@@ -134,6 +146,17 @@ EngineContext createSandbox(final ServiceContext serviceContext) {
     return Collections.unmodifiableMap(persistentQueries);
   }
 
+  Set<String> getQueriesWithSink(final SourceName sourceName) {
```

Comment:
any reason not to make this a `Set<QueryId>`? I feel that keeping this strongly typed is a benefit, especially if we want to leverage `QueryId` more heavily in the future

Reply:
No reason. It's actually better your suggestion.
Done.